### ISSUE TITLE
rubocop -a cause logic error with Style/SoleNestedConditional cop on version 1.27 + ruby 3.1

### ISSUE THREAD

**Issue by zw963** [rubocop/rubocop#10529]
Following is a sample.

```rb
x = 100
y = true
z = true
foo = true

if x.instance_of? Integer
  if !y && !z
    return puts 'ok' if foo
  end
end
```

After run rubocop -a or used with solargraph, it will be change to:

```rb
x = 100
y = true
z = true
foo = true

return puts 'ok' if x.instance_of? Integer && (!y && !z) && foo
```

That was not expected, it should be:

```rb
x = 100
y = true
z = true
foo = true

return puts 'ok' if (x.instance_of? Integer) && (!y && !z) && foo
```

**Comment by zw963** [comment]
Thanks you for quick fixs, i consider this is a safe cops, and will cause imperceptible error when run `rubocop -a`, so if consider release a patch version this or next week?
